### PR TITLE
Add contract tests to CI

### PR DESCRIPTION
**Pull request by mythmon**
* [x] Refactor circle.yml
* [x] Run live server
* [x] Run contract tests against live server

@relud f? on the bash here?

### CODE REVIEW

**Comment by relud** [comment]
lgtm: bin/ci/docker-run

seems to have lost a trailing h: bin/deploy/dockerhub.sh → bin/ci/deploy-dockerhub.s

**Comment by mythmon** [comment]
Ok, this is passing and running all the tests.

@Osmose Can you look at the JS and Python changes and the general way this works in Circle?
@relud Can you double check my bash and Docker changes?

**Comment by relud** [comment]
bash lgtm

File: contract-tests/conftest.py [hunk 1]
```diff
@@ -0,0 +1,44 @@
...
+        dest="server",
+        default="http://localhost:8000",
+        help="Server to run tests against"
+    )
+    parser.addoption(
+        "--no-verify",
+        action="store_false",
+        dest="verify",
+        default=None,
+        help="Server to run tests against"
```
Comment (Osmose) [inline contract-tests/conftest.py]: Wrong help text.

File: contract-tests/conftest.py [hunk 2]
```diff
@@ -0,0 +1,44 @@
...
+
+
+@pytest.fixture
+def conf(request):
+    return request.config
+
+
+class RequestVerifyWrapper(object):
+    """
+    `request.Session` doesn't have a way to permanently change the verify flag. This provides that.
```
Comment (Osmose) [inline contract-tests/conftest.py]: Untrue, they have a `verify` attribute that can be set to false:

```python
session = requests.Session()
session.verify = False
session.get('https://localhost:8000')
```

I tested this in my normandy virtualenv and it seemed to work.

Comment (mythmon) [inline contract-tests/conftest.py]: Ah, nice. I tried `requests.Session(verify=False)`, and then consulted the docs and didn't find anything and gave up.

File: contract-tests/test_api.py [hunk 3]
```diff
@@ -1,26 +1,9 @@
-import configparser
 import jsonschema
 import pytest
-import requests
 
 
-@pytest.fixture
-def conf():
-    config = configparser.ConfigParser()
-    config.read('manifest.ini')
```
Comment (Osmose) [inline contract-tests/test_api.py]: You should remove `manifest.ini` too.

File: contract-tests/conftest.py (hunk 2 above)
Comment (Osmose) [inline contract-tests/conftest.py]: Yeah, it felt like a weird omission so I gave the source code a quick search to find it.

**Comment by Osmose** [comment]
r+wc, everything looks good to me. Nice work!

**Pull request merged**